cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ModelUtils.java

```diff
@@ -188,16 +190,18 @@ public static Secret buildSecret(Reconciliation reconciliation, ClusterCa cluste
             data.put(keyCertName + ".password", certAndKey.storePasswordAsBase64String());
         }
 
-        return createSecret(secretName, namespace, labels, ownerReference, data);
+        return createSecret(secretName, namespace, labels, ownerReference, data, emptyMap(), emptyMap());
     }
 
-    public static Secret createSecret(String name, String namespace, Labels labels, OwnerReference ownerReference, Map<String, String> data) {
+    public static Secret createSecret(String name, String namespace, Labels labels, OwnerReference ownerReference,
+                                      Map<String, String> data, Map<String, String> customAnnotations, Map<String, String> customLabels) {
         if (ownerReference == null) {
             return new SecretBuilder()
                     .withNewMetadata()
                         .withName(name)
                         .withNamespace(namespace)
-                        .withLabels(labels.toMap())
+                        .withLabels(Util.mergeLabelsOrAnnotations(labels.toMap(), customLabels))
+                    .withAnnotations(customAnnotations)
```

Comment:
Can you please indent `withAnnotations`? If you will implement Jakub's suggestion about `createJmxSecret`, ignore this comment.

Reply:
Maybe I misunderstood something but even following Jakub's suggestion we gonna still have this line in this method right ?

Reply:
Yeah, m idea to have the different method in `AbstractModel`, but still call from there this method. So I think this should stay, yes.